### ISSUE TITLE
SubsPlease indexer with Sonarr returns 404 for all releases

### ISSUE THREAD

**Issue by celynw** [autobrr/autobrr#849]
**Version**
1.23.0

**Describe the bug**
The SubsPlease indexer does not seem to parse the URL properly. I could be wrong about the reason but the fact is, it's not working although the releases are still being listed as I expect. I didn't see much in the way to configure that I could have missed.

**To Reproduce**
1. Add built-in SubsPlease indexer. This should automatically create the 'Rizon' IRC entry. Assuming that goes well, there's a green pulsating indicator and it's authenticated properly.
2. Add a filter which uses this SubsPlease indexer. I set the name field in "Movies/Shows" to `*`, and I'll let Sonarr or Radarr handle the matching.
3. Wait for a release and check the releases page. Every single one is rejected. Example:

```text
Rejected: sonarr: failed to push release: {[SubsPlease] One Piece - 1058 (720p) [124AD7F8] https://nyaa.si/view/1661151/torrent  707470000 subsplease torrent torrent 2023-04-16T03:01:32+01:00}: could not push release to sonarr: sonarr: bad request:  (status: 404 Not Found): {
  "message": "NotFound"
}
```
The URL in that message _does_ work. I looked at the source and it looks like the parsing isn't matching?
https://github.com/autobrr/autobrr/blob/9f86c3ab4b7ff07a67a289b3686124ba2cbf5185/internal/indexer/definitions/subsplease.yaml#L47

**Expected behavior**
No 404 error when the URL in the message is in fact a real and valid URL.

**Screenshots**
![image](https://user-images.githubusercontent.com/3299161/232572113-703c7ffb-5f01-4537-8b66-298ff65df1ac.png)
![image](https://user-images.githubusercontent.com/3299161/232571896-aec64bac-cb07-4741-b222-65d0fba691bf.png)

**Desktop (please complete the following information):**
 - OS: Docker image: [`ghcr.io/alexbelgium/autobrr-aarch64:1.23.0`](https://ghcr.io/alexbelgium/autobrr-aarch64:1.23.0)

**Comment by martylukyy** [comment]
Please check if your sonarr client is configured correctly in autobrr.

The error `sonarr: bad request: (status: 404 Not Found): { "message": "NotFound" }`
occurs when autobrr tries to send a release to sonarr, but the web server reports `404 Not Found` as the response.
This is because the web server being contacted here cannot find the requested resource (`/api/v3/release/push`) 
to which autobrr is trying to send the release to, because of it not being sonarr most likely.

**Comment by celynw** [comment]
Ah, thank you for the clarification. It wasn't clear to me what part of the chain was encountering the 404.

I was about to reply that actually that address actually does exist. If I copied the configured host and appended that address it reaches a JSON response.

But actually, my issue was caused by the 'URL base' in sonarr. My real address was actually `$host/$base/api...`. In the above example, it was automatically redirecting to this address. For some reason, each time I update sonarr it repopulates that field, which is why I didn't notice.

Not a bug with autobrr, closing!
